pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -412,21 +458,32 @@ private void uploadSegment(File indexDir, SegmentMetadata segmentMetadata, File
           Response.Status.NOT_FOUND);
     }
 
-    // Check quota
-    StorageQuotaChecker.QuotaCheckerResponse quotaResponse;
-    try {
-      quotaResponse = checkStorageQuota(indexDir, segmentMetadata, offlineTableConfig);
-    } catch (InvalidConfigException e) {
-      // Admin port is missing, return response with 500 status code.
-      throw new ControllerApplicationException(LOGGER, "Quota check failed for segment: " + segmentName + " of table: " + offlineTableName + ", reason: "
-          + e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR);
-    }
-    if (!quotaResponse.isSegmentWithinQuota) {
-      throw new ControllerApplicationException(LOGGER,
-          "Quota check failed for segment: " + segmentName + " of table: " + offlineTableName + ", reason: "
-              + quotaResponse.reason, Response.Status.FORBIDDEN);
+    // Check quota when segment file is locally downloaded
+    // For metadata-only upload, we have no way to check quota because we don't send the file itself and
+    // we can't access length of zipped files in a remote filesystem
```

Comment:
so then maybe the fs can have a getLength API as optional implementation?

Reply:
Agree, can add that in a future one